tests/integrationtest/r/executor/show.result

```diff
@@ -1141,3 +1141,51 @@ t	CREATE TABLE `t` (
   `b` int DEFAULT NULL,
   PRIMARY KEY (`a`) /*T![clustered_index] CLUSTERED */
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin /*T! PRE_SPLIT_REGIONS=2 */
+create database test_show_columns_db;
+create user test_show_columns_user;
+create table test_show_columns_db.t (a int, b int);
+// no grant would report error
+show columns from test_show_columns_db.t;
+Error 1142 (42000): SELECT command denied to user 'test_show_columns_user'@'%' for table 't'
+// with column privilege, specific columns are shown.
+grant select(a) on test_show_columns_db.t to test_show_columns_user;
+show columns from test_show_columns_db.t;
+Field	Type	Null	Key	Default	Extra
+a	int	YES		NULL	
+grant insert(b) on test_show_columns_db.t to test_show_columns_user;
+show columns from test_show_columns_db.t;
+Field	Type	Null	Key	Default	Extra
+a	int	YES		NULL	
+b	int	YES		NULL	
+// with table privilege, all columns are shown.
+revoke select(a),insert(b) on test_show_columns_db.t from test_show_columns_user;
+grant update on test_show_columns_db.t to test_show_columns_user;
+show columns from test_show_columns_db.t;
+Field	Type	Null	Key	Default	Extra
+a	int	YES		NULL	
+b	int	YES		NULL	
+drop database test_show_columns_db;
+drop user test_show_columns_user;
+create database test_show_index_db;
+create user test_show_index_user;
+create table test_show_index_db.t (a int, b int);
+alter table test_show_index_db.t add index i1(b);
+alter table test_show_index_db.t add index i2(b);
+// no grant would report error
+show index from test_show_index_db.t;
+Error 1142 (42000): SELECT command denied to user 'test_show_index_user'@'%' for table 't'
+// with any column privilege, indexes are shown.
+grant update(a) on test_show_index_db.t to test_show_index_user;
+show index from test_show_index_db.t;
+Table	Non_unique	Key_name	Seq_in_index	Column_name	Collation	Cardinality	Sub_part	Packed	Null	Index_type	Comment	Index_comment	Visible	Expression	Clustered	Global
+t	1	i1	1	b	A	0	NULL	NULL	YES	BTREE			YES	NULL	NO	NO
+t	1	i2	1	b	A	0	NULL	NULL	YES	BTREE			YES	NULL	NO	NO
+// with table privilege, indexes are shown.
+revoke update(a) on test_show_index_db.t from test_show_index_user;
+grant references on test_show_index_db.t to test_show_index_user;
+show index from test_show_index_db.t;
+Table	Non_unique	Key_name	Seq_in_index	Column_name	Collation	Cardinality	Sub_part	Packed	Null	Index_type	Comment	Index_comment	Visible	Expression	Clustered	Global
+t	1	i1	1	b	A	0	NULL	NULL	YES	BTREE			YES	NULL	NO	NO
+t	1	i2	1	b	A	0	NULL	NULL	YES	BTREE			YES	NULL	NO	NO
+drop database test_show_index_db;
+drop user test_show_index_user;
```

Comment:
Do we also have tests that cover the DDL operations, like add/drop index where we just have privilege for some columns ...

Reply:
Most of the DDL operations require table privileges like `INDEX` for `CREATE INDEX`.
Check `func (b *PlanBuilder) buildDDL` for more details.

Reply:
So is this the same behavior as MySQL, modify index only check `CREATE_INDEX`, no matter whether the index references a non-privilege column?